ni_measurementlink_service/session_management.py

```diff
@@ -224,6 +294,115 @@ def unreserve(self) -> None:
         """Unreserve sessions."""
         self._session_manager._unreserve_sessions(self._session_info)
 
+    @contextlib.contextmanager
+    def _cache_session(self, session_name: str, session: TSession) -> Generator[None, None, None]:
+        if session_name in self._session_cache:
+            raise RuntimeError(f"Session '{session_name}' already exists.")
+        self._session_cache[session_name] = session
+        try:
+            yield
+        finally:
+            del self._session_cache[session_name]
+
+    @contextlib.contextmanager
+    def _create_session_core(
+        self,
+        session_constructor: Callable[[SessionInformation], TSession],
+        instrument_type_id: str,
```

Comment:
I had the impression we were going to make `instrument_type_id` optional or pass empty string to get 'all' sessions, or in this case, 'the only' session. Am I misremembering? Similar for `_create_sessions_core`.

Reply:
@pbirkhol-ni Are you supporting instrument_type_id="" in the LabVIEW Create Session(s) API?

Reply:
BTW, this parameter only exists in the generic functions. The driver-specific functions know what instrument type id to use.

Reply:
`instrument_type_id` is exposed via the `IInstrument` interface, where there is a VI that you must implement to specify the instrument type id. Right now, we don't do any checking of the value and so a blank string is allowed. However, if you specify a blank string, we will not create any sessions and will not throw an error (at least in the version I am looking at now). For Get Connections, I am returning all connections if the instrument type id is blank. However, now that I think of it, maybe that isn't the right thing to do. Maybe we should error.